server/routes/referrals/referralFormPresenter.ts

```diff
@@ -1,118 +1,123 @@
 /* eslint max-classes-per-file: 0 */
 import DraftReferral from '../../models/draftReferral'
-import ServiceCategory from '../../models/serviceCategory'
+import Intervention from '../../models/intervention'
 
 export default class ReferralFormPresenter {
   private readonly taskValues: typeof ReferralFormPresenter.TaskValues.prototype
 
   private readonly sectionValues: typeof ReferralFormPresenter.SectionValues.prototype
 
-  constructor(private readonly referral: DraftReferral, private readonly serviceCategories: ServiceCategory[]) {
+  private readonly formSectionBuilder: typeof ReferralFormPresenter.FormSectionBuilder.prototype
+
+  constructor(private readonly referral: DraftReferral, private readonly intervention: Intervention) {
     this.taskValues = new ReferralFormPresenter.TaskValues(referral)
     this.sectionValues = new ReferralFormPresenter.SectionValues(this.taskValues)
+    this.formSectionBuilder = new ReferralFormPresenter.FormSectionBuilder(
+      referral,
+      intervention,
+      this.taskValues,
+      this.sectionValues
+    )
   }
 
   get isCohortIntervention(): boolean {
-    return this.serviceCategories.length > 1
+    return this.intervention.serviceCategories.length > 1
   }
 
   get sections(): ReferralFormSectionPresenter[] {
-    const reviewServiceUserInformation: ReferralFormSingleListSectionPresenter = {
-      type: 'single',
-      title: 'Review service user’s information',
-      number: '1',
-      status: this.calculateStatus(this.sectionValues.reviewServiceUserInformation),
-      tasks: [
-        {
-          title: 'Confirm service user’s personal details',
-          url: 'service-user-details',
-        },
-        {
-          title: 'Service user’s risk information',
-          url: this.calculateTaskUrl('risk-information', this.taskValues.serviceUserDetails),
-        },
-        {
-          title: 'Service user’s needs and requirements',
-          url: this.calculateTaskUrl('needs-and-requirements', this.taskValues.riskInformation),
-        },
-      ],
-    }
-    let referralDetails: ReferralFormSectionPresenter
     if (this.isCohortIntervention) {
-      referralDetails = {
-        type: 'multi',
-        title: 'Add intervention referral details',
+      return this.formSectionBuilder.buildCohortReferralSections()
+    }
+    return this.formSectionBuilder.buildSingleReferralSections()
+  }
+
+  private static FormSectionBuilder = class {
```

Comment:
ah now i see what those type definitions were above. what's the reason for having these anonymous class member vars here, vs just having proper classes so we can use the types without having to use `typeof`?

Reply:
I was trying to get around the eslint rule for `max-classes-per-file`, but I've disabled the rule now for this file, so I'll change them to classes to make it a bit more readable.

Reply:
they could always go in their own files too.

Reply:
changed